src/map/script.c

```diff
@@ -8616,6 +8616,53 @@ static BUILDIN(delitem2)
 	return false;
 }
 
+/**
+ * Deletes item at given index.
+ * delitem(<index>{, <amount{, <account id>}});
+ */
+static BUILDIN(delitemidx)
+{
+	struct map_session_data *sd;
+
+	if (script_hasdata(st, 4)) {
+		if ((sd = script->id2sd(st, script_getnum(st, 4))) == NULL) {
+			st->state = END;
+			return true;
+		}
+	} else {
+		if ((sd = script->rid2sd(st)) == NULL)
+			return true;
+	}
+
+	int i = script_getnum(st, 2);
+	if (i < 0 || i >= sd->status.inventorySize) {
+		ShowError("buildin_delitemidx: Index (%d) should be from 0-%d.\n", i, sd->status.inventorySize - 1);
+		st->state = END;
+		return false;
+	}
+
+	int nameid = sd->status.inventory[i].nameid;
```

Comment:
Is this part necessary to check item exists in Item DB?

Reply:
you are right, shouldn't check for it
if the player has unknown item ID in their inventory,
server will display UNKNOWN_ITEM_ID, which is Apple
and this script command will fail to delete that item

in other words, yes delitem has to check for valid item ID,
then this script command will be special that can delete invalid item <-- mention this in doc